conflictlog/pool.go

```diff
@@ -17,16 +17,49 @@ const (
 	DefaultPoolReapInterval = 120 * time.Second
 )
 
+var _ ConnPool = (*connPool)(nil)
+
+// ConnPool defines the behaviour of a connection pool for objects/resources
+// which implements io.Closer interface. The pool should reap the unused resources by
+// calling io.Closer.Close() governed by the GC & reap interval
+type ConnPool interface {
```

Comment:
Should `ConnPool` interface be moved to under `service_def` package?

Reply:
Some of the generic and common things will be rationalized and moved out in subsequent milestones